include/buffer_control_block.h

```diff
@@ -64,14 +70,26 @@ class BufferControlBlock {
    * @param size the size in bytes of the data to write
    * @return true if buffer needs flush and false otherwise
    */
-  bool write(BufferTree *bf, char *data, uint32_t size);
-
-  // some basic functions for access and manipulating a control block
-  inline bool is_leaf()                  {return min_key == max_key;}
-  inline void reset(File_Pointer npos=0) {storage_ptr = npos;}
-  inline buffer_id_t get_id()            {return id;}
-  inline File_Pointer size()             {return storage_ptr;}
-  inline File_Pointer offset()           {return file_offset;}
+  bool write(GutterTree *bf, char *data, uint32_t size);
+
+  // synchronization functions. Should be called when root buffers are read or written to.
+  // Other buffers should not require synchronization
+  // We keep two locks to allow writes to root to happen while flushing
+  // but need to enforce that flushes are done sequentially
+  void lock_flush()   {RW_lock.lock(); flush_lock.lock();}
```

Comment:
Is there some reason why lock_flush(), unlock_flush(), lock_rw(), unlock_rw() aren't inlined?

Reply:
They are now